crates/anvil/src/eth/backend/db.rs

```diff
@@ -385,14 +389,131 @@ impl MaybeFullDatabase for StateDb {
     }
 }
 
+/// Legacy block environment from before v1.3.
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub struct LegacyBlockEnv {
+    pub number: Option<StringOrU64>,
+    #[serde(alias = "coinbase")]
+    pub beneficiary: Option<Address>,
+    pub timestamp: Option<StringOrU64>,
+    pub gas_limit: Option<StringOrU64>,
+    pub basefee: Option<StringOrU64>,
+    pub difficulty: Option<StringOrU64>,
+    pub prevrandao: Option<B256>,
+    pub blob_excess_gas_and_price: Option<LegacyBlobExcessGasAndPrice>,
+}
+
+/// Legacy blob excess gas and price structure from before v1.3.
+#[derive(Debug, Deserialize)]
+pub struct LegacyBlobExcessGasAndPrice {
+    pub excess_blob_gas: u64,
+    pub blob_gasprice: u64,
+}
+
+/// Legacy string or u64 type from before v1.3.
+#[derive(Debug, Deserialize)]
+#[serde(untagged)]
+pub enum StringOrU64 {
+    Hex(String),
+    Dec(u64),
+}
+
+impl StringOrU64 {
+    pub fn to_u64(&self) -> Option<u64> {
+        match self {
+            Self::Dec(n) => Some(*n),
+            Self::Hex(s) => s.strip_prefix("0x").and_then(|s| u64::from_str_radix(s, 16).ok()),
+        }
+    }
+
+    pub fn to_u256(&self) -> Option<U256> {
+        match self {
+            Self::Dec(n) => Some(U256::from(*n)),
+            Self::Hex(s) => s.strip_prefix("0x").and_then(|s| U256::from_str_radix(s, 16).ok()),
+        }
+    }
+}
+
+/// Converts a `LegacyBlockEnv` to a `BlockEnv`, handling the conversion of legacy fields.
+impl TryFrom<LegacyBlockEnv> for BlockEnv {
+    type Error = &'static str;
+
+    fn try_from(legacy: LegacyBlockEnv) -> Result<Self, Self::Error> {
+        Ok(Self {
+            number: legacy.number.and_then(|v| v.to_u256()).unwrap_or(U256::ZERO),
+            beneficiary: legacy.beneficiary.unwrap_or(Address::ZERO),
+            timestamp: legacy.timestamp.and_then(|v| v.to_u256()).unwrap_or(U256::ONE),
+            gas_limit: legacy.gas_limit.and_then(|v| v.to_u64()).unwrap_or(u64::MAX),
+            basefee: legacy.basefee.and_then(|v| v.to_u64()).unwrap_or(0),
+            difficulty: legacy.difficulty.and_then(|v| v.to_u256()).unwrap_or(U256::ZERO),
+            prevrandao: legacy.prevrandao.or(Some(B256::ZERO)),
+            blob_excess_gas_and_price: legacy
+                .blob_excess_gas_and_price
+                .map(|v| BlobExcessGasAndPrice::new(v.excess_blob_gas, v.blob_gasprice))
+                .or_else(|| {
+                    Some(BlobExcessGasAndPrice::new(0, BLOB_BASE_FEE_UPDATE_FRACTION_PRAGUE))
+                }),
+        })
+    }
```

Comment:
lazily default to https://github.com/bluealloy/revm/blob/3ad31163f9fcbcc194db3ed9ac4fc28b822f5848/crates/context/src/block.rs#L104-L120 values